base/applications/mspaint/dib.cpp

```diff
@@ -142,148 +142,146 @@ GetDIBHeight(HBITMAP hBitmap)
+    }
     else
-        LoadString(g_hinstExe, IDS_DEFAULTFILENAME, g_szFileName, _countof(g_szFileName));
+    {
+        ::LoadStringW(g_hinstExe, IDS_DEFAULTFILENAME, g_szFileName, _countof(g_szFileName));
+    }
 
     // set title
     CString strTitle;
     strTitle.Format(IDS_WINDOWTITLE, PathFindFileName(g_szFileName));
     mainWindow.SetWindowText(strTitle);
 
     // update file info and recent
-    g_isAFile = isFile;
+    g_isAFile = isAFile;
     if (g_isAFile)
         registrySettings.SetMostRecentFile(g_szFileName);
 
     g_imageSaved = TRUE;
+}
+
+HBITMAP InitializeImage(LPCWSTR name, LPWIN32_FIND_DATAW pFound, BOOL isFile)
+{
+    COLORREF white = RGB(255, 255, 255);
+    HBITMAP hBitmap = CreateColorDIB(registrySettings.BMPWidth, registrySettings.BMPHeight, white);
+    if (hBitmap == NULL)
+        return NULL;
+
+    HDC hScreenDC = ::GetDC(NULL);
+    g_xDpi = ::GetDeviceCaps(hScreenDC, LOGPIXELSX);
+    g_yDpi = ::GetDeviceCaps(hScreenDC, LOGPIXELSY);
+    ::ReleaseDC(NULL, hScreenDC);
+
+    return SetBitmapAndInfo(hBitmap, name, pFound, isFile);
+}
 
+HBITMAP SetBitmapAndInfo(HBITMAP hBitmap, LPCWSTR name, LPWIN32_FIND_DATAW pFound, BOOL isFile)
+{
+    // update image
+    imageModel.PushImageForUndo(hBitmap);
+    imageModel.ClearHistory();
+
+    SetFileInfo(name, pFound, isFile);
+    g_imageSaved = TRUE;
     return hBitmap;
 }
 
-HBITMAP DoLoadImageFile(HWND hwnd, LPCTSTR name, BOOL fIsMainFile)
+HBITMAP DoLoadImageFile(HWND hwnd, LPCWSTR name, BOOL fIsMainFile)
 {
     // find the file
     WIN32_FIND_DATA find;
-    HANDLE hFind = FindFirstFile(name, &find);
-    if (hFind == INVALID_HANDLE_VALUE)
+    HANDLE hFind = ::FindFirstFileW(name, &find);
+    if (hFind == INVALID_HANDLE_VALUE) // does not exist
     {
-        // does not exist
-        CStringW strText;
-        strText.Format(IDS_LOADERRORTEXT, name);
-        MessageBoxW(hwnd, strText, NULL, MB_ICONERROR);
+        ShowError(IDS_LOADERRORTEXT, name);
         return NULL;
     }
-    DWORD dwFileSize = find.nFileSizeLow; // get file size
-    FindClose(hFind);
+    ::FindClose(hFind);
 
     // is file empty?
-    if (dwFileSize == 0)
+    if (find.nFileSizeLow == 0 && find.nFileSizeHigh == 0)
     {
         if (fIsMainFile)
-        {
-            FILETIME ft;
-            FileTimeToLocalFileTime(&find.ftLastWriteTime, &ft);
-            FileTimeToSystemTime(&ft, &g_fileTime);
-            return SetBitmapAndInfo(NULL, name, dwFileSize, TRUE);
-        }
+            return InitializeImage(name, &find, TRUE);
     }
 
     // load the image
     CImageDx img;
-    img.LoadDx(name, &g_xDpi, &g_yDpi);
+    float xDpi, yDpi;
+    HRESULT hr = img.LoadDx(name, &xDpi, &yDpi);
+    if (FAILED(hr))
+    {
+        ShowError(IDS_LOADERRORTEXT, name);
+        return NULL;
+    }
+
+    HBITMAP hBitmap = img.Detach();
+    if (!fIsMainFile)
+        return hBitmap;
 
+    g_xDpi = xDpi;
+    g_yDpi = yDpi;
     if (g_xDpi <= 0)
         g_xDpi = 96;
```

Comment:
Hardcoded? Why not using `::GetDeviceCaps(hScreenDC, LOGPIXELSX);`etc. as done in the InitializeImage function?

Reply:
It will be fixed in #5404.